server/tso.go

```diff
@@ -82,17 +82,14 @@ func (s *Server) syncTimestamp() error {
 		return errors.Trace(err)
 	}
 
-	var now time.Time
-
-	for {
-		now = time.Now()
-		if wait := subTimeByWallClock(last, now) + updateTimestampGuard; wait > 0 {
-			tsoCounter.WithLabelValues("sync_wait").Inc()
-			log.Warnf("wait %v to guarantee valid generated timestamp", wait)
-			time.Sleep(wait)
-			continue
-		}
-		break
+	now := time.Now()
+	// If the current system time is less than the saved etcd timestamp,
+	// which means that the system time may be incorrect,
+	// the timestamp allocation will start from the saved etcd timestamp temporarily.
+	if subTimeByWallClock(last, now)+updateTimestampGuard > 0 {
+		log.Errorf("system time may be incorrect, last: %v, now: %v", last, now)
+		s.ts.Store(&atomicObject{physical: last})
+		return nil
```

Comment:
I think you want to update `now` here. Or the incorrect system time will be saved.